src/AWS.Deploy.CLI/Commands/GenerateDeploymentProjectCommand.cs

```diff
@@ -0,0 +1,185 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using AWS.Deploy.Common;
+using AWS.Deploy.Orchestration;
+using AWS.Deploy.Orchestration.CDK;
+using AWS.Deploy.Orchestration.Data;
+using AWS.Deploy.Orchestration.Utilities;
+using AWS.Deploy.Recipes;
+
+namespace AWS.Deploy.CLI.Commands
+{
+    public class GenerateDeploymentProjectCommand
+    {
+        private readonly IToolInteractiveService _toolInteractiveService;
+        private readonly IConsoleUtilities _consoleUtilities;
+        private readonly IOrchestratorInteractiveService _orchestratorInteractiveService;
+        private readonly ICdkProjectHandler _cdkProjectHandler;
+        private readonly ICDKManager _cdkManager;
+        private readonly IAWSResourceQueryer _awsResourceQueryer;
+        private readonly IDeploymentBundleHandler _deploymentBundleHandler;
+        private readonly ICommandLineWrapper _commandLineWrapper;
+        private readonly OrchestratorSession _session;
+
+        public GenerateDeploymentProjectCommand(
+            IToolInteractiveService toolInteractiveService,
+            IConsoleUtilities consoleUtilities,
+            IOrchestratorInteractiveService orchestratorInteractiveService,
+            ICdkProjectHandler cdkProjectHandler,
+            ICDKManager cdkManager,
+            IAWSResourceQueryer awsResourceQueryer,
+            IDeploymentBundleHandler deploymentBundleHandler,
+            ICommandLineWrapper commandLineWrapper,
+            OrchestratorSession session)
+        {
+            _toolInteractiveService = toolInteractiveService;
+            _consoleUtilities = consoleUtilities;
+            _orchestratorInteractiveService = orchestratorInteractiveService;
+            _cdkProjectHandler = cdkProjectHandler;
+            _cdkManager = cdkManager;
+            _awsResourceQueryer = awsResourceQueryer;
+            _deploymentBundleHandler = deploymentBundleHandler;
+            _commandLineWrapper = commandLineWrapper;
+            _session = session;
+        }
+
+        public async Task ExecuteAsync(string saveDirectory)
```

Comment:
Is this full path or just the directory name? Some documentation might help.

Reply:
How are we going to test this method? specially file IO given there is quite a bit of logic involved. You might want to use `IFileManager` and `IDirectoryManager` to make this command mockable.

Reply:
I have added integration tests for this command that performs and end-to-end test of all the file IO.